Misc. changes

martinwe-adfinis opened the pull request:
Some commits marked with `[TMP]`, still needs some clean-up.

martinwe-adfinis committed:
etc/sh: Add alias for fixing input upon attaching office equipment

This should preferrably happen automatically/triggered, and ideally also
cover attaching Wacom tablets. But it's "good enough" for now.

martinwe-adfinis committed:
Add shellcheckrc for allowing external sources

martinwe-adfinis committed:
Ignore etc/evince

martinwe-adfinis committed:
etc/vim: Show window-local modified state in inactive window

If we use &modified as a reference, we only get the focused window's
modified state, which leads to the wrong indicator being shown.

The double space is because... honestly, I have no idea. A single space
had no effect, so I randomly tried with a second one, and now it works.

If anyone is able to explain this to me, please please please do so.

martinwe-adfinis committed:
etc/vim: Show normal state in inactive window statusbar

While probably a bit pointless, this makes the transition less jaggy.

The assumption is likely wrong, but I have yet to encounter a case where
I left (or re-entered) a window in another state than normal.

martinwe-adfinis committed:
etc/sh/interactive: Add workaround for recent Groff change

For reference: https://bbs.archlinux.org/viewtopic.php?id=287185

martinwe-adfinis committed:
Stop ignoring share/applications

Currently an empty MIME cache, and I want to keep it that way (or at
least track what's going on there).

ayekat commented on bin/dotbloat:
```diff
@@ -61,7 +61,7 @@ if ! ls -1d --color=auto .[!.]*; then
 
 # automatic decisions for specific files/directories:
-keeplist='.anthy .local .pki .ssh .zshenv'
+keeplist='.anthy .doom.d .emacs.d .local .pki .ssh .vault-token .zshenv .GA_ClientID .ICAClient'
```
This should be removed from the PR. I'm using neither Doom, Emacs, Vault nor Citrix.

ayekat replied:
Dotbloat using config files might be a potential solution to avoid this sort of issues.

I assume that we will never be able to fully get rid of unwanted applications (and their dotfiles in the homedir) in $enterprise environments, unless we start taking a more radical (e.g. containerised) approach.

martinwe-adfinis replied:
Removed.

ayekat commented on etc/sh/interactive.d/80-aliases.sh:
```diff
@@ -20,6 +20,7 @@ alias watch='watch '
 
 # Random other aliases:
 alias cc='gcc -std=c11 -pedantic -Wall -Wextra -Wbad-function-cast -Wcast-align -Wcast-qual -Wconversion -Wfloat-equal -Wformat=2 -Wlogical-op -Wmissing-declarations -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wwrite-strings'
+alias fixinput='sh $XDG_CONFIG_HOME/karuiwm/profile.d/10-xinput.sh && sh $XDG_CONFIG_HOME/X11/xprofile.d/10-setxkbmap.sh'
```
While marked as "tmp", I would actually like to keep this around here as well. The usecase is common enough that I expect this to happen also in my personal environment.

Sure something more elegant (see #34) might be a better longterm solution, but… welp, longterm solutions also usually take a long time to grow.

martinwe-adfinis replied:
Removed "tmp" marker (and also reworded commit message to reflect that it's an alias, not a script).

ayekat replied:
LGTM

ayekat commented on etc/vim/vimrc:
```diff
@@ -63,6 +63,7 @@ call neobundle#begin(g:bundledir)
 NeoBundleFetch 'Shougo/neobundle.vim'
 
 " Declare used bundles:
+NeoBundle 'axvr/org.vim'
```
Again, please remove this (I believe this is part of the same Emacs/Doom commit as the other change).

martinwe-adfinis replied:
Removed.

ayekat commented on share/.gitignore:
```diff
@@ -23,3 +22,6 @@
 # Gnome:
 /evolution/
 /gvfs-metadata/
+
+# Emacs:
+/fonts/
```
… and also remove this Emacs-related change.

martinwe-adfinis replied:
Removed.

martinwe-adfinis committed:
etc/qutebrowser: Update Arch Linux manpage search engine

It's now officially part of Arch Linux. :-D

(has been for a while, actually...)

ayekat commented:
Needs rebase.

martinwe-adfinis commented:
> Needs rebase.

Rebased.
